pkg/cmd/enclave_secrets.go

```diff
@@ -97,13 +97,15 @@ func init() {
 	enclaveSecretsCmd.Flags().StringP("project", "p", "", "enclave project (e.g. backend)")
 	enclaveSecretsCmd.Flags().StringP("config", "c", "", "enclave config (e.g. dev)")
 	enclaveSecretsCmd.Flags().Bool("raw", false, "print the raw secret value without processing variables")
+	enclaveSecretsCmd.Flags().Bool("visibility", false, "when printing the table, include secret visibility types")
```

Comment:
The JSON output will always include the visibility for computed, and will include visibility for raw when `--raw` is specified (which is why my prior wording on the visibility flag was narrowed to the table output). `--visibility` has no impact on the JSON output.

I opted for this because we're now potentially returning `null` in the JSON output for restricted secrets, and I wanted the visibility to be present to explain that behavior. Do you disagree?

Reply:
That's tough, I could go either way. I think your point is valid and can get behind your current implementation.